### ISSUE TITLE
`gem_package` resource doesn't accept [anymore] pessimistic version constraints

### ISSUE THREAD

**Issue by saveriomiroddi** [chef/chef#7768]
## Description
`gem_package` resource doesn't accept [anymore] pessimistic version constraints

## Chef Version
14.5.33

## Platform Version
Ubuntu Server 18.04

## Replication Case
Execute the `gem_package` resource as specified in the section below.

## Client Output
```
  * gem_package[aws-sdk] action install

    ================================================================================
    Error executing action `install` on resource 'gem_package[aws-sdk]'
    ================================================================================

    ArgumentError
    -------------
    Malformed version number string ~> 3.0.1

    Resource Declaration:
    ---------------------
    # In /path/to/recipes/default.rb

     45:     gem_package gem_name do
     46:       # not relevant
     47:       version '~> 3.0.1'
     48:     end
     49:   end

    Compiled Resource:
    ------------------
    # Declared in /path/to/recipes/default.rb:45:in `block in from_file'

    gem_package("aws-sdk") do
      package_name "aws-sdk"
      action [:install]
      default_guard_interpreter :default
      declared_type :gem_package
      cookbook_name "ruby"
      recipe_name "default"
      version "~> 3.0.1"
      gem_binary "/usr/bin/gem"
    end

    System Info:
    ------------
    chef_version=14.5.33
    platform=ubuntu
    platform_version=18.04
    ruby=ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux]
    program_name=/usr/bin/chef-client
    executable=/opt/chef/bin/chef-client
```

**Comment by saveriomiroddi** [comment]
Some extra information: if a gem satistyfing the versioning is already installed (eg. 3.0.1 in this case), the exception is not raised.

**Comment by lamont-granquist** [comment]
fixed already in #7703 and  #7705 in 14.6.x which will be released any day now...
